packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol

```diff
@@ -226,6 +296,9 @@ contract FuelERC20GatewayV4 is
         uint8 decimals = _getTokenDecimals(tokenAddress);
         uint256 amount = _adjustWithdrawalDecimals(decimals, l2BurntAmount);
 
+        // rate limit check
+        _addWithdrawnAmount(tokenAddress, amount);
```

Comment:
Wouldn't we achieve significant gas savings if we added a check for the existence of a rate limiter?

Reply:
We cannot do that, as we cannot initialize a rate limiter for every token that will ever exist. By default we must use no rate limiter